src/util/generate_root.c

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+#include <string.h>
+#include <xtt.h>
+
+int generate_root(struct cli_params* params)
+{
+    const char *privkey_filename = params->privkey;
+    const char *pubkey_filename = params->pubkey;
+    const char *id_filename = params->id;
+    char *root_pub_file = "root_pub.bin";
+    char *root_priv_file = "root_priv.bin";
+    xtt_ecdsap256_pub_key pub;
+    xtt_ecdsap256_priv_key priv;
+    xtt_certificate_root_id root_id;
+
+    unsigned char idpubkey[sizeof(xtt_ecdsap256_pub_key)+sizeof(xtt_certificate_root_id)];
+
+    // 0) Read root id from file, giving root_id a random number if there is no file given
+    if (id_filename != NULL) {
+        assert(-1 != read_from_file(id_filename, root_id.data, sizeof(xtt_certificate_root_id)));
```

Comment:
`assert` only exists when you build in debug, so all these statements that are in `assert` actually won't exist when you build in release!

`assert` in C/C++ is almost a documentation, something you're saying to someone else reading your code "This should always be true." It should only be used to check things that are set in the code itself (i.e. to check your own assumptions about, for example, some constants you defined in another file), and not to check anything that might depend on outside environment (i.e. if the user gave you stupid input, or if the OS is out of memory).

So, these should just be typical return value checks using `if`.